earth_enterprise/src/common/serverdb/serverdbReader.cpp

```diff
@@ -395,9 +443,21 @@ const MimeType ServerdbReader::GetData(
       }
     } else if (request == "Json") {
       std::string json_variable_name = arg_map["var"];
-      // CheckJavascriptName(json_variable_name);
-      GetJson(json_variable_name, language, region, buf);
-      return MapTileUtils::kJavascriptMimeType;
+      if (!arg_map["v"].empty()) {
+        // CheckJavascriptName(json_variable_name);
+        if (arg_map["v"] == "2") {
+          GetJson(json_variable_name, language, region, buf);
+        } else if (arg_map["v"] == "1") {
+          // This is added for backwards compatibility reasons.
+          GetJsonV1(json_variable_name, language, region, buf);
+        } else {
+          throw khSimpleException("Illegal argument found in v parameter: ") << arg_map["v"];
```

Comment:
I would change `Illegal` to `Invalid`. May be the following error message could be an alternative: `Invalid value for the the v parameter: ... A valid value must be either 1 or 2.`

Reply:
@williamstrong you still need to address this comment.

Reply:
@tst-rwildes thank you for pointing this out. I didn't see it on the main page of the PR.